src/analysis/complex/locally_uniform_limit.lean

```diff
@@ -0,0 +1,184 @@
+
+lemma cderiv_eq_deriv (hU : is_open U) (hf : differentiable_on ℂ f U) (hr : 0 < r)
+  (hzr : closed_ball z r ⊆ U) :
+  cderiv r f z = deriv f z :=
+two_pi_I_inv_smul_circle_integral_sub_sq_inv_smul_of_differentiable hU hzr hf (mem_ball_self hr)
+
+lemma norm_cderiv_le (hr : 0 < r) (hf : ∀ w ∈ sphere z r, ‖f w‖ ≤ M) :
+  ‖cderiv r f z‖ ≤ M / r :=
+begin
+  have hM : 0 ≤ M,
+  { obtain ⟨w, hw⟩ : (sphere z r).nonempty := normed_space.sphere_nonempty.mpr hr.le,
+    exact (norm_nonneg _).trans (hf w hw) },
+  have h1 : ∀ w ∈ sphere z r, ‖((w - z) ^ 2)⁻¹ • f w‖ ≤ M / r ^ 2,
+  { intros w hw,
+    simp only [mem_sphere_iff_norm, norm_eq_abs] at hw,
+    simp only [norm_smul, inv_mul_eq_div, hw, norm_eq_abs, map_inv₀, complex.abs_pow],
+    exact div_le_div hM (hf w hw) (sq_pos_of_pos hr) le_rfl },
+  have h2 := circle_integral.norm_integral_le_of_norm_le_const hr.le h1,
+  simp only [cderiv, norm_smul],
+  refine (mul_le_mul le_rfl h2 (norm_nonneg _) (norm_nonneg _)).trans (le_of_eq _),
+  field_simp [_root_.abs_of_nonneg real.pi_pos.le, real.pi_pos.ne.symm, hr.ne.symm],
+  ring
+end
+
+lemma cderiv_sub (hr : 0 < r) (hzr : closed_ball z r ⊆ U)
+  (hf : continuous_on f U) (hg : continuous_on g U) :
+  cderiv r (f - g) z = cderiv r f z - cderiv r g z :=
+begin
+  have h1 : continuous_on (λ (w : ℂ), ((w - z) ^ 2)⁻¹) (sphere z r),
+  { refine ((continuous_id'.sub continuous_const).pow 2).continuous_on.inv₀ (λ w hw h, hr.ne _),
+    rwa [mem_sphere_iff_norm, sq_eq_zero_iff.mp h, norm_zero] at hw },
+  simp_rw [cderiv, ← smul_sub],
+  congr' 1,
+  simp_rw [pi.sub_apply, smul_sub],
+  refine circle_integral.integral_sub _ _,
+  { have := (h1.smul (continuous_on.mono hf ((sphere_subset_closed_ball).trans hzr))),
+    exact continuous_on.circle_integrable hr.le this },
+  { have := (h1.smul (continuous_on.mono hg ((sphere_subset_closed_ball).trans hzr))),
+    exact continuous_on.circle_integrable hr.le this }
+end
+
+lemma norm_cderiv_lt (hr : 0 < r) (hzr : closed_ball z r ⊆ U) (hfM : ∀ w ∈ sphere z r, ‖f w‖ < M)
+  (hf : continuous_on f U) :
+  ‖cderiv r f z‖ < M / r :=
+begin
+  obtain ⟨L, hL1, hL2⟩ : ∃ L < M, ∀ w ∈ sphere z r, ‖f w‖ ≤ L,
+  { have e1 : sphere z r ⊆ U := sphere_subset_closed_ball.trans hzr,
+    have e2 : (sphere z r).nonempty := normed_space.sphere_nonempty.mpr hr.le,
+    have e3 : continuous_on (λ w, ‖f w‖) (sphere z r),
+      from continuous_norm.comp_continuous_on (hf.mono e1),
+    obtain ⟨x, hx, hx'⟩ := (is_compact_sphere z r).exists_forall_ge e2 e3,
+    exact ⟨‖f x‖, hfM x hx, hx'⟩ },
+  exact (norm_cderiv_le hr hL2).trans_lt ((div_lt_div_right hr).mpr hL1)
+end
+
+lemma norm_cderiv_sub_lt (hr : 0 < r) (hzr : closed_ball z r ⊆ U)
+  (hfg : ∀ w ∈ sphere z r, ‖f w - g w‖ < M) (hf : continuous_on f U) (hg : continuous_on g U) :
+  ‖cderiv r f z - cderiv r g z‖ < M / r :=
+cderiv_sub hr hzr hf hg ▸ norm_cderiv_lt hr hzr hfg (hf.sub hg)
+
+lemma tendsto_uniformly_on_cderiv (hf : continuous_on f U) (hδ : 0 < δ)
```

Comment:
do you really need `hf` in this lemma? Can you deduce the relevant fact from `hF`? Maybe you don't need `hf` but it's not worth removing it as this is just a helper lemma on the way to some more important theorem -- in this case, you can mention that in a docstring to the lemma.

Reply:
Done - in fact I rewrote the previous lemmas so that they do not involve `U` at all, it makes the statements more transparent.